internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,278 @@
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	err := wait.PollImmediateUntil(time.Duration(200*time.Millisecond), podCheck, ctx.Done())
+	if err != nil {
+		return fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	return err
+}
+
+func (rp *RegistryPod) init() error {
+	rp.setDefaults()
+	err := rp.validate()
+	if err != nil {
+		return fmt.Errorf("error in validating registry pod: %v", err)
+	}
+	return nil
+}
+
+func (rp *RegistryPod) setDefaults() {
+	if rp.GRPCPort == 0 {
+		rp.GRPCPort = defaultGRPCPort
+	}
+
+	if len(rp.IndexImage) == 0 {
+		rp.IndexImage = defaultIndexImage
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		if rp.IndexImage == defaultIndexImage {
+			rp.BundleAddMode = "semver"
+		} else {
+			rp.BundleAddMode = "replaces"
+		}
+	}
+}
+
+func (rp *RegistryPod) validate() error {
+	if len(rp.BundleImage) == 0 {
+		return errors.New("bundle image is required")
+	}
+	if len(rp.DBPath) == 0 {
+		return errors.New("registry database path is required")
+	}
+
+	if len(rp.Namespace) == 0 {
+		return errors.New("pod namespace is required")
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		panic("bundle add mode is not set")
+	}
+
+	return nil
+}
+
+// getPodName will return a string constructed from the bundle Image name
+func getPodName(bundleImage string) (string, error) {
+	image, err := container.ImageFromString(bundleImage)
+	if err != nil {
+		return "", fmt.Errorf("invalid bundle image name: %s", bundleImage)
+	}
+
+	return image.Name + "-index", nil
+}
+
+// build sets the defaults and validates the RegistryPod struct, and returns a pod definition
+func (rp *RegistryPod) build() (*corev1.Pod, error) {
+	podName, err := getPodName(rp.BundleImage)
+	if err != nil {
+		return nil, fmt.Errorf("error in building pod: %v", err)
+	}
+
+	containerCmd, err := rp.getContainerCmd()
+	if err != nil {
+		return nil, fmt.Errorf("error in parsing container command: %v", err)
+	}
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: rp.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  defaultContainerName,
+					Image: rp.IndexImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						containerCmd,
+					},
+					Ports: []corev1.ContainerPort{
+						{Name: defaultContainerPortName, ContainerPort: rp.GRPCPort},
+					},
+				},
+			},
+		},
+	}
+
+	return pod, nil
+}
+
+func (rp *RegistryPod) getContainerCmd() (string, error) {
+	const containerCommand = `/bin/mkdir -p {{ .DBPath | basename }} &&  /bin/opm registry add -d {{ .DBPath | basename }} -b {{.BundleImage}} --mode={{.BundleAddMode}} && /bin/opm registry serve -d {{ .DBPath | basename }} -p {{.GRPCPort}}`
```

Comment:
Required to scroll the line here which make very harder understanding the code and go against the good practices. 
This is exactly the kind of situation that the lint avoids and I am afraid to start to come back to be introduced again. 

Could you please break this line?